src/test/scala/sigmastate/utils/ByteReaderWriterImpSpecification.scala

```diff
@@ -138,7 +141,26 @@ class ByteReaderWriterImpSpecification extends PropSpec
     checkSize(4398046511104L, 562949953421311L, 7)
     checkSize(562949953421312L, 72057594037927935L, 8)
     checkSize(72057594037927936L, Long.MaxValue, 9)
-    checkSize(Long.MinValue, 0, 10)
+    checkSize(Long.MinValue, -1L, 10)
+  }
+
+  private def bytesZigZaggedLong(v: Long): Array[Byte] =
+    byteArrayWriter().putULong(encodeZigZagLong(v)).toBytes
+
+  private def checkSizeZigZagged(low: Long, high: Long, size: Int): Unit = {
+    // Gen.choose does not always include range limit values
+    bytesZigZaggedLong(low).length shouldBe size
+    bytesZigZaggedLong(high).length shouldBe size
+    forAll(Gen.choose(low, high)) { v: Long =>
+      bytesZigZaggedLong(v).length shouldBe size
+    }
+  }
+
+  property("size of serialized zigzag'ed data") {
+    // source: http://github.com/scodec/scodec/blob/055eed8386aa85ff27dba3f72b104a8aa3d6012d/unitTests/src/test/scala/scodec/codecs/VarLongCodecTest.scala#L16
+    checkSizeZigZagged(-64L, 63L, 1)
+    checkSizeZigZagged(-8192L, -65L, 2)
+    checkSizeZigZagged(64L, 8191L, 2)
```

Comment:
@greenhat what about other ranges up to 9 bytes?

Reply:
@ergomorphic No excuses for being lazy. :) Fixed.